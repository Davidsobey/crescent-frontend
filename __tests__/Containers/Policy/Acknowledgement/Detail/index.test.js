import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

// Component to be tested
import UsersPolicyDetails from '../../../../../src/Containers/Policy/Acknowledgement/Detail/index.js';


const mockStore = configureStore();
const initialState = {
  PolicyReducer:{
    policyMaterials: [],
    policyMaterials_loading: false,
    policyId: 1,
    policyName: 'name',
    policyDescription: 'description',
    canAcknowlege: true,
    acknowledging: false,
  },
  LoginReducer:{
    user: {},
  },
};
const store = mockStore(initialState);

describe('<UsersPolicyDetails />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<UsersPolicyDetails store={store} />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});