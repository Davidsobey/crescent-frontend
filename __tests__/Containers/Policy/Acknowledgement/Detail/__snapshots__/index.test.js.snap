// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersPolicyDetails /> render() renders the component 1`] = `
<Connect(Form(UsersPolicyDetails))
  canAcknowlege={true}
  destroyOnUnmount={true}
  dispatch={[Function]}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="userView"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  policyDescription="description"
  policyId={1}
  policyMaterials={Array []}
  policyMaterials_loading={false}
  policyName="name"
  policy_acknowledging={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
  user={Object {}}
/>
`;
